core/meta/object-descriptor.js

```diff
@@ -357,35 +358,121 @@ var ObjectDescriptor = exports.ObjectDescriptor = Montage.specialize( /** @lends
     },
 
 
-    _propertyDescriptors: {
-        value: null
-    },
+    _arrayToSet: {
```

Comment:
Similar code is in pipeline-converter.js, we can use Set.from() that Collections provide

Reply:
Will update to use `new Set(anArray)` as well.